xla/python/xla.cc

```diff
@@ -274,9 +278,26 @@ NB_MODULE(xla_extension, m_nb) {
         auto tcp_device = gloo::transport::tcp::CreateDevice(tcp_attrs);
         return std::make_shared<cpu::GlooCollectives>(std::move(gloo_kv_store),
                                                       std::move(tcp_device));
+#elifdef  __APPLE__
+        std::shared_ptr<KeyValueStoreInterface> kv_store = nullptr;
+        if (distributed_client != nullptr) {
+          kv_store = GetDistributedKeyValueStore(distributed_client,
+                                                 /*key_prefix=*/"cpu:");
+        }
+        auto gloo_kv_store = std::make_unique<cpu::GlooKeyValueStore>(kv_store);
+        auto uv_attrs = gloo::transport::uv::attr();
+        if (hostname) {
+          uv_attrs.hostname = *hostname;
+        }
+        if (interface) {
+          uv_attrs.iface = *interface;
+        }
+        auto uv_device = gloo::transport::uv::CreateDevice(uv_attrs);
+        return std::make_shared<cpu::GlooCollectives>(std::move(gloo_kv_store),
+                                                      std::move(uv_device));
 #else   // __linux__
         throw xla::XlaRuntimeError(
-            "make_gloo_tcp_collectives only implemented for linux");
+            "make_gloo_tcp_collectives only implemented for linux and macos");
```

Comment:
We probably should rename the method to just `make_gloo_collectives`. But I'll handle this later since this involves changing JAX as well.

Reply:
Sounds good! Note that if the `libuv` dependency is okay to add (I wasn't sure about that, might be tricky in `google3`?), it's possible to use it for Linux too which would allow merging the code paths.

Reply:
Yes, `libuv` is okay for google3. On the open source side, I think it's fine to add one more dependency. Cc'ing @ddunl just in case.

Reply:
Excellent! In that case, the code in this file can probably be simplified to look more more or less like they did after https://github.com/openxla/xla/pull/7674, except for using the `uv` protocol instead of the `tcp` one.